lib/src/chunking.rs

```diff
@@ -364,78 +380,447 @@ impl Chunking {
     }
 }
 
-type ChunkedComponents<'a> = Vec<&'a ObjectSourceMetaSized>;
-
+#[cfg(test)]
 fn components_size(components: &[&ObjectSourceMetaSized]) -> u64 {
     components.iter().map(|k| k.size).sum()
 }
 
 /// Compute the total size of a packing
 #[cfg(test)]
-fn packing_size(packing: &[ChunkedComponents]) -> u64 {
+fn packing_size(packing: &[Vec<&ObjectSourceMetaSized>]) -> u64 {
     packing.iter().map(|v| components_size(v)).sum()
 }
 
-fn sort_packing(packing: &mut [ChunkedComponents]) {
-    packing.sort_by(|a, b| {
-        let a: u64 = components_size(a);
-        let b: u64 = components_size(b);
-        b.cmp(&a)
+fn mean(data: &[u64]) -> Option<f64> {
+    let sum = data.iter().sum::<u64>() as f64;
+    let count = data.len();
+    match count {
+        positive if positive > 0 => Some(sum / count as f64),
+        _ => None,
+    }
+}
+
+fn std_deviation(data: &[u64]) -> Option<f64> {
+    match (mean(data), data.len()) {
+        (Some(data_mean), count) if count > 0 => {
+            let variance = data
+                .iter()
+                .map(|value| {
+                    let diff = data_mean - (*value as f64);
+                    diff * diff
+                })
+                .sum::<f64>()
+                / count as f64;
+            Some(variance.sqrt())
+        }
+        _ => None,
+    }
+}
+
+fn median_absolute_deviation(data: &mut Vec<u64>) -> (f64, f64) {
+    //Sort data
+    //data.sort_by(|a, b| a.partial_cmp(b).unwrap());
+
+    //Find median of data
+    let median_data: f64 = match data.len() % 2 {
+        1 => data[data.len() / 2] as f64,
+        _ => 0.5 * (data[data.len() / 2 - 1] + data[data.len() / 2]) as f64,
+    };
+
+    //Absolute deviations
+    let mut absolute_deviations = Vec::new();
+    for size in data {
+        absolute_deviations.push(f64::abs(*size as f64 - median_data))
+    }
+
+    absolute_deviations.sort_by(|a, b| a.partial_cmp(b).unwrap());
+    let mad: f64 = match absolute_deviations.len() % 2 {
+        1 => absolute_deviations[absolute_deviations.len() / 2],
+        _ => {
+            0.5 * (absolute_deviations[absolute_deviations.len() / 2 - 1]
+                + absolute_deviations[absolute_deviations.len() / 2])
+        }
+    };
+
+    (median_data, mad)
+}
+
+//Assumes components is sorted by descending size
+//Use MAD as threshold to partition packages [abs(low_limit), high_limit]
+fn get_partitions_with_threshold(
```

Comment:
This one could se a doc comment that describes what it does at a high level.
Also in Rust, official "documentation comments" use `///`